crates/holochain/src/conductor/conductor.rs

```diff
@@ -433,22 +443,29 @@ where
         let root_env_dir = std::path::PathBuf::from(self.root_env_dir.clone());
         let keystore = self.keystore.clone();
 
-        let cells_tasks = cell_ids_with_proofs.into_iter().map(|(cell_id, proof)| {
-            let root_env_dir = root_env_dir.clone();
-            let keystore = self.keystore.clone();
-            let conductor_handle = conductor_handle.clone();
-            let cell_id_inner = cell_id.clone();
-            tokio::spawn(async move {
-                let env = EnvironmentWrite::new(
-                    &root_env_dir,
-                    EnvironmentKind::Cell(cell_id_inner.clone()),
-                    keystore.clone(),
-                )?;
-                Cell::genesis(cell_id_inner, conductor_handle, env, proof).await
-            })
-            .map_err(CellError::from)
-            .and_then(|result| async move { result.map(|_| cell_id) })
-        });
+        let cells_tasks = cell_ids_with_proofs
+            .into_iter()
+            .map(|(cell_id, proof)| async {
+                let root_env_dir = root_env_dir.clone();
+                let keystore = self.keystore.clone();
+                let conductor_handle = conductor_handle.clone();
+                let cell_id_inner = cell_id.clone();
+                let ribosome = conductor_handle
+                    .get_ribosome(cell_id.dna_hash())
+                    .await
+                    .map_err(Box::new)?;
+                tokio::spawn(async move {
+                    let env = EnvironmentWrite::new(
+                        &root_env_dir,
+                        EnvironmentKind::Cell(cell_id_inner.clone()),
+                        keystore.clone(),
+                    )?;
+                    Cell::genesis(cell_id_inner, conductor_handle, env, ribosome, proof).await
+                })
+                .map_err(CellError::from)
+                .and_then(|result| async move { result.map(|_| cell_id) })
+                .await
```

Comment:
This seems to we spawning and then immediately awaiting. Perhaps you just want to await directly without the spawn?

Reply:
This is a collection of tasks that gets executed in parallel later. Also I didn't write this in this PR.